rmf_task_msgs/msg/Phase.msg

```diff
@@ -0,0 +1,24 @@
+# level of the phase tree, from 1, 2....n
+int32 tree_level
+
+# type of the current phase
+uint32 current_phase
+
+# type of the remaining phases
+uint32[] pending_phases
+
+# completed phases type 
+uint32[] completed_phases
+
+uint32 PHASE_GOTOPLACE=1
```

Comment:
How would you feel about providing two fields per phase:

1. Short generic title (e.g. "Receive Item")
2. Detailed description (e.g. "[coke_bot_#42] dispenses [Coke Zero]")

The short generic titles could have some convention for a maximum character length while the detailed description can be arbitrarily long and give complete details.

Reply:
If we want to get super fancy we could come up with a detail text format that the UI could use to highlight and/or hyperlink text items.

For example:

```
[dispenser:coke_bot_#42] dispenses [item:Coke Zero]
```

could be parsed to

```
coke_bot_#42 dispenses Coke Zero
```

but where `coke_bot_#42` and `Coke Zero` are highlighted in a special way. Maybe hovering over the special text can bring up a tooltip with more details on the robot or item, and clicking on one of them could open a panel specific to the robot or item.

Reply:
> How would you feel about providing two fields per phase:
> 
> 1. Short generic title (e.g. "Receive Item")
> 2. Detailed description (e.g. "[coke_bot_#42] dispenses [Coke Zero]")
> 
> The short generic titles could have some convention for a maximum character length while the detailed description can be arbitrarily long and give complete details.

Yeah, this will be very helpful.



> hyperlink text items.
> 
> For example:
> 
> ```
> [dispenser:coke_bot_#42] dispenses [item:Coke Zero]
> ```
> 
> could be parsed to
> 
> ```
> coke_bot_#42 dispenses Coke Zero
> ```
> 
> but where `coke_bot_#42` and `Coke Zero` are highlighted in a special way. Maybe hovering over the special text can bring up a tooltip with more details on the robot or item, and clicking on one of them could open a panel specific to the robot or item.

I don't see any problems if the formatting is kept simple enough that it is still human readable. I don't think this is something in the roadmap of rmf-web though, but in any case it will still be a string so there won't be changes to the message structure.

Reply:
Currently, the "detailed description" is being populated [in taskSummary](https://github.com/open-rmf/rmf_internal_msgs/blob/ecb86fb2e4e941cc1b12449bf1af1d2716b4850b/rmf_task_msgs/msg/TaskSummary.msg#L21). I guess in this case, we can move it to the `PhaseTier` s, each tier will have a `description` field for verbose description.

Reply:
I think the `status` message should be changed to something that describes if the task is proceeding correctly. E.g. `Okay`, or `Error: Blocked by [magni_#66]`.

Reply:
Updated https://github.com/open-rmf/rmf_internal_msgs/pull/14/commits/b9ffda31d65d16f8db0aeabcf64dbd3c4942c612